CRM/Core/BAO/Email.php

```diff
@@ -71,6 +71,12 @@ public static function create($params) {
 
     $email->save();
 
+    $contactId = (int) ($email->contact_id ?? CRM_Core_DAO::getFieldValue(__CLASS__, $email->id, 'contact_id'));
+    if ($contactId && $email->is_primary) {
```

Comment:
If it is changed from primary to not primary no update will happen but I think that is ok because what is set in place of this will trigger a change